gpMgmt/bin/go-tools/cli/init.go

```diff
@@ -156,10 +194,42 @@ func InitClusterServiceFn(inputConfigFile string, force, verbose bool) error {
 	}
 
 	err = ParseStreamResponse(stream)
+
+	//Call the cleanup routine only if the file exists
+	fileName := filepath.Join(Conf.LogDir, constants.CleanFileName)
+	_, errS := utils.System.Stat(fileName)
+
 	if err != nil {
-		return err
+
+		if errS == nil {
+			//Log the error received from makeCluster
+			gplog.Error("Cluster creation failed, error %v", err)
+			gplog.Error("There has been an error during cluster creation.\n")
+			gplog.Error("Would you like to rollback? Enter 'yes' or 'no': \n")
+
+			input := GetUserInput()
+			if input {
+				//The user has asked to delete the datadirectories
+				//Call Hub rpc for cleanup
+				_, errS := HubClient.CleanCluster(context.Background(), &idl.CleanClusterRequest{})
+				if errS != nil {
+					return utils.LogAndReturnError(fmt.Errorf("clean cluster command failed: %v", errS))
+				} else {
+					gplog.Info("clean cluster command successful")
+					return nil
+				}
+
+			} else {
```

Comment:
Remove the else statement and make it default for better readability.
```suggestion
				} 
				gplog.Info("clean cluster command successful")
				return nil
			} else {
```